### ISSUE TITLE
[Bug]: Shaders decompilation broken

### ISSUE THREAD

**Issue by CREAsTIVE** [AssetRipper/AssetRipper#1229]
### Are you on the latest version of AssetRipper?

Yes, I'm on the latest release of AssetRipper.

### Which release are you using?

Windows x64

### Which game did this occur on?

[Slime Rancher](https://steamdb.info/app/433340/)

### Which Unity version did this occur on?

2019.4.29f1

### Is the game Mono or IL2Cpp?

Mono

### Describe the issue.

So simmilar to [this discussion](https://github.com/AssetRipper/AssetRipper/discussions/643) i got same error. Cameras just wont work if PostProcess is enabled. However, in my case, in addition to the postprocessor shaders, 90% of the regular shaders also broken. Their code look approximately the same:
```shaderlab
Shader "SR/Paintlight/Triplanar Topper" {
	Properties {
		// A lot of (diffrent and correct for each shader) shader properties
	}
	//DummyShaderTextExporter
	SubShader{
		Tags { "RenderType" = "Opaque" }
		LOD 200
		CGPROGRAM
#pragma surface surf Standard
#pragma target 3.0
		struct Input
		{
			float2 uv_MainTex;
		};

		void surf(Input IN, inout SurfaceOutputStandard o)
		{
			o.Albedo = 1;
		}
		ENDCG
	}
	Fallback "Diffuse"
	//CustomEditor "ShaderForgeMaterialInspector"
}
```
And of course nothing renders correctly:
![image](https://github.com/AssetRipper/AssetRipper/assets/62616308/bb5c4841-9544-4777-b31a-bfeff1391960)

some Post-process shaders also broken, but in a different way. So for example content of `Hidden/SSAO Pro V2`:
```shaderlab
Shader "Hidden/SSAO Pro V2" {
	Properties {
		_MainTex ("", 2D) = "white" {}
		_SSAOTex ("", 2D) = "white" {}
		_NoiseTex ("", 2D) = "white" {}
	}
	//DummyShaderTextExporter
	SubShader{
		Tags { "RenderType"="Opaque" }
		LOD 200
		CGPROGRAM
#pragma surface surf Standard
#pragma target 3.0

		sampler2D _MainTex;
		struct Input
		{
			float2 uv_MainTex;
		};

		void surf(Input IN, inout SurfaceOutputStandard o)
		{
			fixed4 c = tex2D(_MainTex, IN.uv_MainTex);
			o.Albedo = c.rgb;
			o.Alpha = c.a;
		}
		ENDCG
	}
}
```
its should contains atleast 7 passes (but in reality its dont, what cause post-process errors)

### Relevant log output

[AssetRipper.log](https://github.com/AssetRipper/AssetRipper/files/14466812/AssetRipper.log)

**Comment by ds5678** [comment]
Shader decompilation is provided "as-is" because I do not work on that section of the project, and the individual who contributed the feature has higher priorities and has given no indication that their priorities will change anytime soon.

**Comment by CREAsTIVE** [comment]
@ds5678 Sad... But is there a way to import and use shaders without decompilation?

**Comment by ds5678** [comment]
Dummy and yaml are both good options

**Comment by EXtremeExploit** [comment]
Same problem on My Little Pony: A Maretime bay Adventure, Unity version is 2020.3.26f1 it uses URP, Toony Colors 2 Pro and shader graph shaders. Exact same shader issue with mostly everything being white

**Comment by AmogusJS** [comment]
> Same problem on My Little Pony: A Maretime bay Adventure, Unity version is 2020.3.26f1 it uses URP, Toony Colors 2 Pro and shader graph shaders. Exact same shader issue with mostly everything being white

Same problem here with the same app and the toony colors shader. It looks like the shader graph was used to apply the texture to the material which now doesn't load because the shader is replaced with a dummy file and everything defaults to grey. It still should be possible to retexture everything and I'm currently looking for a possible fix.

Did you manage to solve it by any chance?

**Comment by EXtremeExploit** [comment]
Nope, as said above the current shader decomp is broken on this unity version at this moment, all we can do is wait or fix it ourselves but i dont have the knowledge to do it

**Comment by JxxIT** [comment]
Any updates so far?

**Comment by ds5678** [comment]
@JxxIT The situation has slightly changed, but not significantly. After the first iteration of IL2Cpp Recovery is finished, I plan to look into shader decompilation.

**Comment by JxxIT** [comment]
@ds5678 That was fast! Great, do you have an ETA or anything I can temporarily use?

**Comment by ds5678** [comment]
@JxxIT I'm not working on shader decompilation right now, so there's no ETA for it.

**Comment by elthundercloud** [comment]
Is it even possible to decompile shaders into even a remotely readable format?

**Comment by ds5678** [comment]
@elthundercloud readability is not one of the goals for decompilation. The two goals are high accuracy of output and strong robustness against diversity of input.
